modules/metalsmith/doc_assets.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+/* eslint-env es6 */
+
+
+const {info, error} = require('./../debugger')('nx_assets');
+const fs = require('fs');
+const crypto = require('crypto');
+const Nuxeo = require('nuxeo');
+const cheerio = require('cheerio');
+
+const map = {
+    // 'a'     : 'href',
+    // 'link'  : 'href',
+    // 'script': 'src',
+    'img'  : 'src',
+    'video': 'src',
+    'audio': 'src'
+    // 'source': 'src'
+};
+
+
+const checksum = function (str, algorithm, encoding) {
+    return crypto
+        .createHash(algorithm || 'md5')
+        .update(str, 'utf8')
+        .digest(encoding || 'hex');
+};
+
+const get_attribute = function getAttribute (el) {
+    return (typeof map[el.name] !== 'undefined')
+        ? map[el.name]
+        : false;
+};
+
+const is_html = function isHTML (file) {
+    return /\.html?/.test(file);
+};
+
+
+const check_file = (files, filename, selector, nuxeo) => {
+    return new Promise(function (resolve) {
+
+        const file = files[filename];
+
+        const contents = file.contents.toString();
+        const $ = cheerio.load(contents);
+
+        const url_promises = [];
+
+        $(selector).each(function (i, el) {
```

Comment:
I'd probably factor this into functions to have something like:
`Promise.all(
   $(selector)
    .filter(_nxAssetUrls)
    .map(_fetchAsset)
)`
where `_fetchAsset` would return a Promise